source/22_test/index.rst

```diff
@@ -0,0 +1,155 @@
+
+* When the code is updated, you can run unit tests to ensure that regression errors do not occur.
+* Through fine-grained unit test design, you can quickly and accurately locate the source of errors during unit test.
+* Combining unit testing with code coverage ensures that all code and branches have been tested.
+* After finding bugs, you can add test cases that can reproduce bugs to unit tests to continuously improve the perfection of code functions.
+* Another important point -- **for a module, reading unit test code is also a very efficient way to understand its function and usage**.
+
+
+.. _ref-test-types:
+
+Types of Unit Test
+---------------------------------
+
+In the DI-engine project, we divide the unit test into the following parts:
+
+* ``unittest`` -- functional unit test in a general sense to ensure the normal function of engineering code and the convergence of algorithm code on simple use cases.
+* ``algotest`` -- unit test for algorithm code to ensure that the algorithm code can meet the use requirements on specific use cases.
+* ``cudatest`` -- unit test for CUDA dependent features, ensure that such features function normally in the operating environment with CUDA.
+* ``envpooltest`` -- unit test for features that rely on envpool high-performance parallel computing to ensure that such features function normally.
+* ``platformtest`` -- unit test of cross platform code, ensure that the core functions of DI engine can still operate normally on MacOS and Windows platforms.
+* ``benchmark`` -- performance test of algorithm or architecture, speed measurement is mainly carried out for relevant contents to ensure that its performance meets the requirements.
+
+
+
+How to Build Unit Test
+---------------------------------
+
+In DI engine, we use `pytest <https://docs.pytest.org/>`_ to build unit tests.
+
+For unit test writing, you can refer to the ``tests`` folder under the code path at all levels as a whole, such as `ding/envs/env_manager/tests <https://github.com/opendilab/DI-engine/tree/main/ding/envs/env_manager/tests>`_.
+
+
+Naming Conventions
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+For unit testing, we generally build it in the unit of class or function, and its name should meet certain specifications, specifically:
+
+* For the unit test of function form, the function name is required to be started with ``test_``.
+* For the unit test of class form, the class name is required to be started with ``Test``, and names of all methods used for testing should start with ``test_``.
+
+
+Assertions
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+In the test cases, we use ``assert`` (assertion) to check the prototype results. If the assertion is not true, very detailed information will be displayed, as shown in the following figure
+
+.. image:: pytest_assert.png
+    :scale: 55%
+    :align: center
+
+In addition, `` pytest`` also supports assertion of thrown exceptions, as shown below
+
+.. code:: python
+
+   import pytest
+
+   def test_zero_division():
```

Comment:
Add @pytest.mark.unittest in the code example

Reply:
Fixed.